core/ir/aarch64/encode.c

```diff
@@ -248,12 +248,17 @@ instr_encode_arch(dcontext_t *dcontext, instr_t *instr, byte *copy_pc, byte *fin
         } else {
             /* We were unable to encode this instruction. */
             IF_DEBUG({
-                char disas_instr[MAX_INSTR_DIS_SZ];
-                instr_disassemble_to_buffer(dcontext, instr, disas_instr,
-                                            MAX_INSTR_DIS_SZ);
-                SYSLOG_INTERNAL_ERROR("Internal Error: Failed to encode instruction:"
-                                      " '%s'\n",
-                                      disas_instr);
+                /* Avoid complaining for !assert_reachable as we hit failures for
+                 * branches to labels during instrumentation (i#5297).
+                 */
+                if (assert_reachable) {
+                    char disas_instr[MAX_INSTR_DIS_SZ];
+                    instr_disassemble_to_buffer(dcontext, instr, disas_instr,
+                                                MAX_INSTR_DIS_SZ);
+                    SYSLOG_INTERNAL_ERROR("Internal Error: Failed to encode instruction:"
+                                          " '%s'\n",
+                                          disas_instr);
+                }
```

Comment:
Can the encoding failure be due to something other than reachability problems here? In that case we probably don't want to ignore the failure right?

Reply:
The assert_reachable flag tells us whether to make noise or keep silent, so we do want to ignore if it is not set.  We should trust the caller to set it properly.

Reply:
The caller might set `assert_reachable=true` to silence reachability errors, but still want to get notified of other encoding failures.

Reply:
Not sure what you are proposing to be done: if this SYSLOG is printed even when `assert_reachable` is false, we're going to have tons of spurious output (see the issue: every single jump to a label, which for bbdup or memtrace is multiple per block, will print this out).  We don't want that.

Reply:
> we probably don't want to ignore the failure right?

The encoding still fails.  There is just no SYSLOG.  The caller can decide what to do.

Reply:
> Not sure what you are proposing to be done

I assume `assert_reachable` is intended to silence only reachability errors, but not other errors.

If so, it seems ideal if we can find the cause of the encoding failing, and SYSLOG only if `failure_was_not_due_to_reachability || assert_reachable`. This way, the user can still see logs for other errors, even if they have silenced reachability errors.

Not sure how hard it is to find the encoding failure reason. So, LGTM.

Reply:
Agreed, if it were simple, to have the failure reason here: but I think it would take work to try and propagate the reason from inside deeper encoding call chains.  32-bit arm has an attempt at propagating some things; I think the other arches are harder to figure out for misc encoding errors.